internal/tui/scanschedule.go

```diff
@@ -0,0 +1,72 @@
+package tui
+
+import (
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+const (
+	fastScanInterval = 2 * time.Second
+	slowScanInterval = 10 * time.Second
+)
+
+// ScanSchedule is a component that triggers scans at a regular interval.
+type ScanSchedule struct {
+	callback  func() tea.Msg
+	isRunning bool
+	interval  time.Duration
+}
+
+// NewScanSchedule creates a new ScanSchedule.
+func NewScanSchedule(callback func() tea.Msg) *ScanSchedule {
+	return &ScanSchedule{
+		callback: callback,
+		interval: fastScanInterval, // Start with a fast scan
+	}
+}
+
+// Start begins the scan schedule.
+func (s *ScanSchedule) Start() tea.Cmd {
+	s.isRunning = true
+	return s.tick()
+}
+
+// Stop halts the scan schedule.
+func (s *ScanSchedule) Stop() {
+	s.isRunning = false
+}
+
+// Update handles messages for the ScanSchedule.
+func (s *ScanSchedule) Update(msg tea.Msg) tea.Cmd {
+	if !s.isRunning {
+		return nil
+	}
+	switch msg.(type) {
+	case tickMsg:
+		// When we get a tick, call the callback and then schedule the next tick.
+		return tea.Batch(s.callback, s.tick())
+	}
+	return nil
+}
+
+// HasResults should be called when a scan has returned results.
+func (s *ScanSchedule) HasResults(hasResults bool) {
```

Comment:
Let's replace the HasResults, Start, Stop functions with a single function: `SetSchedule(interval time.Duration)` and make a few constants to use with this value:

```go
scanOff = 0 * time.Second
scanFast = 2 * time.Second
scanSlow = 10 * time.Second
```